test/es-module/test-cjs-esm-warn.js

```diff
@@ -14,7 +14,7 @@ const pjson = path.resolve(
   fixtures.path('/es-modules/package-type-module/package.json')
 );
 
-(async () => {
+mustCall(async () => {
```

Comment:
No, as the error message says, you must add `.then(common.mustCall)`. Is there something we can do to make the message clearer? We want the test to catch never settling promises, ensuring the `.then` is called is the way to do it.

Reply:
> The result of an immediately-invoked async function needs to be used (e.g. with `.then(common.mustCall())`)

The whole message is confusing IMHO:

* Now that I understand what to do, "needs to be used" seems misdirection—that makes me thinking of something like tree-shaking.
* The `e.g. with` makes me think it's merely an example, like it should be added to code that was already using `.then()` like `myAsyncThing.then(mustCall((data) => strictEqual(data, 'foo')))`; if it said something like ``Append `.then(mustCall)` to the immediately-invoked async function`` I think there would be no question what to do.

I tried a bunch of permutations before asking, and they all either resulting in the test not running or mustCall erroneously complaining that the test was not called (despite `node:test`'s output clearly showing that it was).

Reply:
So, actually, no. I still don't understand what needs to be done. Appending `.then(mustCall)` satisfies the lint rule but actually fails when run.

Reply:
> Appending `.then(mustCall)` satisfies the lint rule but actually fails when run.

That makes sense, what you meant to write is `.then(mustCall())`, not `.then(mustCall)`.

Reply:
🤦‍♂️🙇‍♂️